### PR TITLE
feat(registration): add error message handling

### PR DESCRIPTION
**Pull request by LorenzoS92**
Add handling of errors during registration via API

### CODE REVIEW

File: src/registered.js [hunk 1]
```diff
@@ -71,7 +71,15 @@ module.exports = async function (request, h) {
...
-      throw `errore imprevisto nel salvataggio, riprovare più tardi (${data?.title}, ${data?.detail})`;
+      let errorMsg = '';
+      if (data.validationErrors) {
+        for (const error of data.validationErrors) {
+          errorMsg += `Valore non valido: <b>${error.value}</b> per il field: <b>${error.field}</b> con la regola: <b>${error.rule}</b></br>`;
+        }
+        throw errorMsg;
+      }
+
+      throw data.detail;
```
Comment (bfabio) [inline src/registered.js:82]:
```suggestion
       throw data.validationErrors?.map(e => `
         Valore non valido: <strong>${error.value}</strong> per il campo: <strong>${error.field}</strong>
         con la regola: <strong>${error.rule}</strong><br>
       `).join('') || data.detail
```

File: src/registered.js [hunk 2]
```diff
@@ -90,13 +98,16 @@ module.exports = async function (request, h) {
...
+        errorMsg: 'Errore imprevisto nel salvataggio, riprovare più tardi',
+        errorDetail: err,
         referente,
         refTel,
         ipa,
         url,
         pec,
         amministrazione,
+        apiError: true,
+        token: token,
```
Comment (bfabio) [inline src/registered.js:110]:
```suggestion
        token,
```

File: src/registered.js (hunk 1 above)
Comment (LorenzoS92) [reply src/registered.js:82]: Yes, thank you. Much better.
I haven't played in javascript for a while 🙁

**Comment by bfabio** [comment]
@LorenzoS92 I'd keep the copy and layout updates in a different PR and have this be just only for the error handling

**Comment by LorenzoS92** [comment]
Ok @bfabio will open a new PR

**Review by bfabio — APPROVED** [review]

**Pull request merged**